pkg/controllers/nodeclaim/lifecycle/liveness.go

```diff
@@ -103,3 +107,16 @@ func (l *Liveness) updateNodePoolRegistrationHealth(ctx context.Context, nodeCla
 	}
 	return nil
 }
+
+func (l *Liveness) deleteNodeClaimForTTL(ctx context.Context, reason string, nodeClaim *v1.NodeClaim) error {
```

Comment:
nit: Do we think we need this helper? Honestly I think we could probably just inline a lot of this

Reply:
Need, no, but without it there's code duplication.